include/core/RestoreMacros.h

```diff
@@ -10,44 +10,60 @@
 namespace ml {
 namespace core {
 
-#define RESTORE(tag, restore)                                                      \
-    if (name == tag) {                                                             \
-        if ((restore) == false) {                                                  \
-            LOG_ERROR(<< "Failed to restore " #tag ", got " << traverser.value()); \
-            return false;                                                          \
-        }                                                                          \
-        continue;                                                                  \
+#define RESTORE(tag, restore)                                                          \
+    if (name == tag) {                                                                 \
+        if ((restore) == false) {                                                      \
+            if (traverser.value().empty()) {                                           \
+                LOG_ERROR(<< "Failed to restore " #tag);                               \
+            } else {                                                                   \
+                LOG_ERROR(<< "Failed to restore " #tag ", got " << traverser.value()); \
```

Comment:
In many cases, i.e. sub-levels, we know that the value will be an empty string. This makes the messages easier to read in this case.